tests/wscript

```diff
@@ -279,9 +279,45 @@ def build(bld):
     # clang on Linux errors on true == true or false == false compile-time assertions
     bld.env.CFLAGS.append('-Wno-tautological-compare')
 
+    # Any test in this list won't be compiled
+    bld.env.BROKEN_TESTS = [
+        'test_app_fetch_endpoint.c',
+        'test_graphics_draw_text_flow.c',
+        'test_perimeter.c',
+        'test_ancs_pebble_actions.c',
+        'test_timeline_actions.c',
+        'test_bluetooth_persistent_storage_prf.c',
+        'test_bluetooth_persistent_storage.c',
+        'test_session.c',
+        'test_session_receive_router.c',
+        'test_compositor.c',
+        'test_floor.c',
+        'test_pow.c',
+        'test_ams.c',
+        'test_ams_util.c',
+        'test_gap_le_advert.c',
+        'test_bt_conn_mgr.c',
+        'test_gatt_client_accessors.c',
+        'test_gatt_client_discovery.c',
+        'test_gatt_client_subscriptions.c',
+        'test_gatt_service_changed_client.c',
+        'test_gatt_service_changed_server.c',
+        'test_gap_le_connect.c',
+        'test_ancs_util.c',
+        'test_ancs.c',
+        'test_kernel_le_client.c',
+        'test_ppogatt.c'
+    ]
+
+    # Don't run the python tool tests because they exercise a lot of old python2 code that still needs to be updated
+    bld.env.PYTHON_TOOL_TESTS_DISABLED = True
+
+    # Disable warning promotion. Not ideal, but gets most of the tests running again without straight up disabling the new warnings
+    bld.env.CFLAGS.append('-Wno-error')
+
     # time_t is defined in sys/types in newlib, and time.h on recent Linux
     # so just force the defined type for testing time
-    bld.env.CFLAGS.append('-Dtime_t=__SYSCALL_SLONG_TYPE')
+    #bld.env.CFLAGS.append('-Dtime_t=__SYSCALL_SLONG_TYPE')
```

Comment:
Can this be removed rather than just commenting, if it isn't needed anymore?

Reply:
Done